npm/pre-install.js

```diff
@@ -1,10 +1,23 @@
+// @ts-check
+import {familySync, GLIBC, MUSL} from "detect-libc"
+import get_matched_platform from "./utils.js"
+
 const os = process.platform
 const arch = process.arch
+const libcFamily = familySync()
+
+let libc = ""
+if (os === "win32") {
+  libc = "msvc"
+} else {
+  libc = libcFamily === GLIBC ? "gnu" : libcFamily === MUSL ? "musl" : ""
+}
```

Comment:
duplicate code

Reply:
yes.
we can make a function inside utils.